src/lld/dpdk/lld_dpdk_lib.c

```diff
@@ -72,16 +73,30 @@ lld_dpdk_pipeline_mirror_config(struct rte_swx_pipeline *p, void *mir_cfg)
 {
 	int rc = BF_SUCCESS;
 	struct rte_swx_pipeline_mirroring_params mir_params;
+	FILE *io_file = NULL;
+	char buffer[BUF_SIZE];
 
 	mir_params.n_slots = ((struct rte_swx_pipeline_mirroring_params *)mir_cfg)->n_slots;
 	mir_params.n_sessions = ((struct rte_swx_pipeline_mirroring_params *)mir_cfg)->n_sessions;
 
-	if (rte_swx_pipeline_mirroring_config(p, &mir_params)) {
-		LOG_ERROR("Could not configure the pipeline mirror params.");
-		rc = BF_UNEXPECTED;
+	io_file = fopen("/tmp/iospec.io", "a+");
```

Comment:
See whether it make sense to add a new procedure which takes the buffer to be appended to IOSPEC file and use the same API in all the places. I can see it can reduce a lot of code as well as provides more readability.

Reply:
taken care